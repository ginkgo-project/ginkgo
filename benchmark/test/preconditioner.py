#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
)

# input matrix file
test_framework.compare_output(
    ["-input_matrix", str(test_framework.matrixpath)],
    expected_stdout="preconditioner.matrix.stdout",
    expected_stderr="preconditioner.matrix.stderr",
)

# set preconditioner works
test_framework.compare_output(
    [
        "-preconditioners",
        "jacobi",
        "-jacobi_max_block_size",
        "32",
        "-jacobi_storage",
        "0,0",
        "-input",
        '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="preconditioner.precond.stdout",
    expected_stderr="preconditioner.precond.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="preconditioner.profile.stdout",
    expected_stderr="preconditioner.profile.stderr",
)

# stdin
test_framework.compare_output(
    ["-reorder", "amd"],
    expected_stdout="preconditioner.reordered.stdout",
    expected_stderr="preconditioner.reordered.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)
