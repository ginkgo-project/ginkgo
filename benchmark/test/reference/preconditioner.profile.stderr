Running on reference(0)
Running with 0 warm iterations and 1 running iterations
The random seed for right hand sides is 42
Running with preconditioners: none
Running test case stencil(100, 7pt)
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill_in_matrix_data
DEBUG: end   dense::fill_in_matrix_data
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill_in_matrix_data
DEBUG: end   dense::fill_in_matrix_data
Matrix is of size (125, 125), 725
DEBUG: begin stencil(100, 7pt)
	Running preconditioner: none
DEBUG: begin none
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: begin repetition generate
DEBUG: begin generate(<typename>)
DEBUG: end   generate(<typename>)
DEBUG: end   repetition generate
DEBUG: begin repetition apply
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: end   repetition apply
DEBUG: end   none
DEBUG: end   stencil(100, 7pt)
